src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -480,12 +506,33 @@ private static bool TryParseGuid(ReadOnlySpan<char> guidString, ref GuidResult r
                         TryParseExactN(guidString, ref result),
             };
         }
+        private static bool TryParseGuid(ReadOnlySpan<byte> guidString, ref GuidResult result)
```

Comment:
Nit: I think this whole method can be made generic and merged with the existing one for `char`. Just found there is a generic method `Number.SpanTrim`.